### PR TITLE
feat: support PKCE for code flow

### PR DESCRIPTION
**Pull request by embbnux**

### CODE REVIEW

**Comment by kirill-konshin** [comment]
LGTM, please add tests.

**Comment by embbnux** [comment]
Thanks, will do it

File: sdk/src/platform/Platform-spec.ts [hunk 1]
```diff
@@ -477,6 +495,14 @@ describe('RingCentral.platform.Platform', () => {
                 ).to.equal(
                     'http://whatever/restapi/oauth/authorize?response_type=code&redirect_uri=http%3A%2F%2Ffoo&client_id=whatever&state=&brand_id=&display=&prompt=&ui_options=&ui_locales=&localeId=',
                 );
+
+                expect(
+                    platform
+                        .loginUrl({
+                            usePKCE: true,
+                        })
+                        .indexOf('code_challenge') > 0,
```
Comment (kirill-konshin) [inline sdk/src/platform/Platform-spec.ts:504]: `expect(...).stringContaining('code_challenge')`: https://jestjs.io/docs/en/expect#expectstringcontainingstring

File: sdk/src/platform/Platform.ts [hunk 2]
```diff
@@ -158,22 +163,54 @@ export default class Platform extends EventEmitter {
...
+            state,
+            brand_id: brandId,
+            display,
+            prompt,
+            ui_options: uiOptions,
+            ui_locales: uiLocales,
+            localeId,
+        };
+        this._codeVerifier = '';
+        if (usePKCE && !implicit) {
```
Comment (kirill-konshin) [inline sdk/src/platform/Platform.ts:190]: Looks like these two flags are mutually exclusive, please throw an error if both are set. And add test for this too.

**Pull request merged**